Fix text colors. Issue #249

IanUme opened the pull request:
**Issue #249** : text color too light to be seen
**Fix** : change the color to a darker color

**Changes**
collect_app/src/main/res/layout/password_dialog_layout.xml
textColor: @android:color/primary_text_dark ->@android:color/darker_gray

collect_app/src/main/res/layout/server_auth_dialog.xml
textColor: @android:color/primary_text_dark ->@android:color/darker_gray

This is my first pull request. If branch names, pull request, or anything else can be improved please let me know!

Commit:
Fix text colors. Issue #249

lognaturel commented:
Welcome and congrats on your first PR @IanUme! Looks great. ✨

This wasn't mentioned in the issue but I'm seeing now that the admin password dialog's buttons don't match the current standard. Could you please make those match the server auth dialog? Might as well get it completely up to date while you're in that code!

Commit:
change dialog to AlertDialog

Commit:
Merge branch 'master' into fix.249

Commit:
change password location

Commit:
remove PasswordDialogPreference

IanUme commented:
Hi @lognaturel. Thanks for the info.

I came across a few problems as changing the buttons was not an arbitrary task. The buttons were part of a layout. Since the project does not use support libraries to my knowledge, it was best to change from a custom DialogPreference to an AlertDialog (to future proof button types). I removed the PasswordDialogPreference class, changed the admin_preference xml and moved the logic to AdminPreferencesActivity.

Well let me know if this works or not. I did some light testing but not an extreme amount (just changed the password a few times and went to the main menu and insured it was changed/removed).

Oh yeah am I supposed to push to my forked repo (I did) or just add commits without pushing? I'm new to this. Thanks!

lognaturel commented on collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesActivity.java:
```diff
@@ -115,6 +123,74 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
+				LayoutInflater factory = LayoutInflater.from(AdminPreferencesActivity.this);
+				final View dialogView = factory.inflate(R.layout.password_dialog_layout, null);
+
+				// Get the server, username, and password from the settings
```
This comment and the line it refers to are not relevant to this dialog!

lognaturel commented on collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesActivity.java:
```diff
@@ -115,6 +123,74 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
+                final EditText passwordEditText = (EditText) dialogView.findViewById(R.id.pwd_field);
+                final EditText verifyEditText = (EditText) dialogView.findViewById(R.id.verify_field);
+
+
```
Extra blank line.

lognaturel reviewed (changes requested):
Thanks, @IanUme! In retrospect, I should have broken the button change into a separate issue and I apologize for rushing to add it on to this PR when I noticed the problem. But what you've done is great! I appreciate that you broke what you did into commits and that you pulled out string resources.

I marked a couple of small changes inline. We are currently in the process of developing style guidelines and a checkstyle file that will enforce it so for now I didn't look too much at it. It'll be something to consider for future PRs.

Commits are local so you do need to push just like you did for others to be able to view your changes. Looks great!

Commit:
Clean up AdminPreferencesActivity.java

Commit:
Merge from upstream

IanUme commented:
Thanks for the suggestions @lognaturel. I made the quick fixes but noticed there were conflicts with the main code, so I pulled from upstream and merged it to my branch (I have no confidence in the merge).

I just looked at the changes and it shows 19000 lines on the merge so I surely failed. The b977047 commit is hopefully okay but there are conflicts with master.

Sorry for any poor coding styles. I didn't see anything in the contributing file when I first read through it and I try to follow Google's Android code guidelines (although I haven't read through it in probably over a year).

lognaturel commented:
You're joining us at a very exciting time, @IanUme. 😊 There's a lot of cleanup of legacy code going on and new processes are being put in place. If you want to be involved or just observe behind the scenes, feel free to join the Slack chat at https://opendatakit.slack.com/

The giant merge was because a PR was just approved to make consistent use of spaces for indentation. The change touched almost every file and you did the merge perfectly!

BUT although I said style wasn't going to be enforced yet since guidelines are still being confirmed, now we will be enforcing spaces for indentation since the codebase is now consistent. So one last thing -- please use spaces for indentation in AdminPreferencesActivity.java (I think that's the only one with tabs). If you're using Android Studio, the easiest thing is just to use Edit > Convert Indents > To Spaces. 

Thanks a million! 🙏

Commit:
Fix indents(?)

IanUme commented:
Hi @lognaturel, I wasn't able to find any tabs in AdminPreferencesActivity.java, I hope I didn't miss it somewhere else. I used Android Studio's auto format (and I have it set to spaces instead of tabs) and pushed the changes.

I'd be happy to join the slack chat but I don't know how.. I tried going to signup and it said the admin has not enabled email signups. I work full-time so I am probably best off staying behind the scenes as I can't reliably code when needed.

yanokwa commented:
Hi @IanUme! The slack URL is http://slack.opendatakit.org and being mostly behind the scenes is just fine!

lognaturel commented:
Oops, I gave you the wrong URL for the Slack, sorry! The one @yanokwa gave is correct. No problem with being mostly a lurker and joining in when you have time or it's a topic you're interested in.

The indentation change worked great. It's hard to notice tabs because they're often set to look like 4 spaces but you can see them clearly in GitHub diffs because they are set to be 8 spaces wide. For example, you can see your latest change in action [here](https://github.com/opendatakit/collect/pull/261/commits/9696e32de4ff7a17bb67806578fd87d703591dca).

@yanokwa, please merge. Congrats and thank you, @IanUme! 🎉🔥🎉
